src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/LocalDBTest/LocalDBTest.cs

```diff
@@ -133,16 +159,20 @@ private static void OpenConnection(string connString)
         }
 
         private static string GetLocalDbNamedPipe()
+        {
+            RestartLocalDB();
+            return ExecuteLocalDBCommandProcess(s_commandPrompt, s_sqlLocalDbInfo, "pipeName");
+        }
+
+        private static void RestartLocalDB()
         {
             string state = ExecuteLocalDBCommandProcess(s_commandPrompt, s_sqlLocalDbInfo, "state");
             while (state.Equals("stopped", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
If the service couldn't start it will be an infinitive loop. Try for 5 times and throw an exception if exceeded.